ext/MitsubaLoader/CElementShapeOBJ.cpp

```diff
@@ -0,0 +1,79 @@
+#include "CElementShapeOBJ.h"
+
+#include "ParserUtil.h"
+#include "CSimpleElement.h"
+
+namespace irr { namespace ext { namespace MitsubaLoader {
+
+
+CElementShapeOBJ::~CElementShapeOBJ()
+{
+	if (mesh)
+		mesh->drop();
+}
+
+bool CElementShapeOBJ::processAttributes(const char** _atts)
+{
+	//only type is an acceptable argument
+	for (int i = 0; _atts[i]; i += 2)
+	{
+		if (std::strcmp(_atts[i], "type"))
+		{
+			ParserLog::wrongAttribute(_atts[i], getLogName());
+			return false;
+		}
+	}
+
+	return true;
+}
+
+bool CElementShapeOBJ::onEndTag(asset::IAssetManager& _assetManager, IElement* _parent)
+{
+	if (!fileName.size())
+	{
+		_IRR_DEBUG_BREAK_IF(true);
+		//print error (name of the .obj file is not set)
+		return false;
+	}
+
+	//getAssetInHierarchy?
```

Comment:
You're gonna be wanting to be using a variant of this one
https://github.com/buildaworldnet/IrrlichtBAW/blob/2bf1c6a3e3518068c12433d560c23c2e9075b0c0/include/irr/asset/IAssetManager.h#L215

hierarchy is as follows:
- 0 for root level asset (an ICPUMesh for now, we have no scene object)
- 1 for whatever ICPUMesh contains, so meshbuffers

So the hierarchy would be 0u, 1u in the future if we make an ICPUScene object or something like that.